label_studio/io_storages/utils.py

```diff
@@ -109,3 +116,61 @@ def parse_range(range_header):
         end = ''
 
     return start, end
+
+
+def _load_tasks_json(blob_str: str, key: str) -> tuple[list[dict], list[int | None], list[int | None]]:
+    """
+    Parse blob_str containing task JSON(s) and return the validated result or raise an error.
+
+    Args:
+        blob_str (str): The blob string to parse.
+        key (str): The key of the blob. Used for error messages.
+
+    Returns:
+        list[dict]: parsed tasks.
+        list[int|None]: row_index for each task.
+        list[int|None]: row_group for each task.
+    """
+
+    def _error_wrapper(exc: Optional[Exception] = None):
+        raise ValueError(
+            (
+                f"Can't import JSON-formatted tasks from {key}. If you're trying to import binary objects, "
+                f'perhaps you\'ve forgot to enable "Treat every bucket object as a source file" option?'
+            )
+        ) from exc
+
+    # TODO: rely on file extensions here instead of pure exception-chaining? Would be more readable, potentially less reliable, would work for all storage types except redis.
```

Comment:
I think exception chaining is actually a better answer than file extensions, because it's not uncommon to use .txt or similar to represent JSON. Would favor removing this comment